tests/unit/geom/CoordinateArraySequenceFactoryTest.cpp

```diff
@@ -79,7 +79,7 @@ namespace tut
 			ensure( 0 != col);
 
 			const size_t size0 = 0;
-			CoordinateSequencePtr sequence = factory->create(col);
```

Comment:
Yes, dim3isM, or "preferM" as I think you used elsewhere...
And, agreed about POINT(NaN NaN) probably only being needed in
the input routines (actually WKB as for WKT it's really "POINT EMPTY").